### PR TITLE
Regrantor new project flow

### PR DESCRIPTION
**Pull request by rachelweinberg12**
When a regranter creates a grant the project immediately gets inserted into the projects row.

If the grant recipient is already on Manifund:
- the project is created under their account
- a transaction is created from the grant maker to the recipient
- the recipient gets an email letting them know they've received a grant

If the grant recipient isn't already on Manifund:
- they recieve an email prompting them to sign up for Manifund with a link to the sign up page (which will have a note telling them to use the email that they recieved the prompt on)
- Once they sign up, a webhook gets called which will check for any pending project transfers. For each one, it'll update the owner of the project, mark the transfer as complete, and add a USD transaction from the regranter to the recipient.

There is a switch on the edit-profile page that controls regranter status, though that isn't actually used anywhere besides for permission to create grants for new projects. As a separate PR I'll add all the front end stuff that makes this actually accessible.

### CODE REVIEW

File: pages/api/create-grant.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  const supabase = createEdgeClient(req)
+  const resp = await supabase.auth.getUser()
+  const user = resp.data.user
+  if (!user) return NextResponse.error()
+  const regranterProfile = await getProfileById(supabase, user.id)
+  if ((toEmail && toUsername) || (!toEmail && !toUsername)) {
+    return NextResponse.error()
+  }
+
+  let slug = title
```
Comment (akrolsmir) [inline pages/api/create-grant.ts:37]: If we're reusing this logic between create-project and create-grant, it's worth pulling these out to a utility `slugify` function somewhere

File: pages/api/create-grant.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    creator: toUsername ? toProfile.id : user.id,
+    title,
+    blurb: subtitle,
+    description,
+    min_funding: amount,
+    funding_goal: amount,
+    founder_portion: TOTAL_SHARES,
+    type: 'grant' as Project['type'],
+    stage: 'active',
+    round: 'Independent', // Create new grants round?
```
Comment (akrolsmir) [inline pages/api/create-grant.ts:63]: Yeah, I would try to have a "Regrants" round. I think specifically it'd be good to run a time-bound test of regranting, maybe ending before July 2023?

File: pages/api/create-grant.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      grant_amount: amount,
+    }
+    const { error: error2 } = await supabase
+      .from('project_transfers')
+      .insert([project_transfer])
+    if (error2) {
+      throw error2
+    }
+  } else {
+    const { error: error3 } = await supabase.from('txns').insert({
```
Comment (akrolsmir) [inline pages/api/create-grant.ts:83]: I found out something cool about supabase recently -- you can skip the `if(error3) {throw error3}` pattern by adding `.throwOnError()` to the end of a supabase call!

File: utils/email.ts [hunk 4]
```diff
@@ -3,35 +3,46 @@ import { SupabaseClient } from '@supabase/supabase-js'
...
-      To: toEmail,
-      TemplateId: templateId,
-      TemplateModel: templateModel,
-    })
-    .then((response) => {
-      console.log('Sending message')
-      console.log(response.To)
-      console.log(response.Message)
-    })
+  // Using fetch instead of postmark's client because it doesn't work on the edge:
```
Comment (akrolsmir) [inline utils/email.ts:27]: note, if we're removing the postmark client we can probably remove it from our packages.json by running `yarn remove postmark`

**Review by akrolsmir — APPROVED** [review]
Thanks for writing such a detailed PR description, it helps me as a reviewer orient myself to the code changes!

File: utils/email.ts (hunk 4 above)
Comment (rachelweinberg12) [reply utils/email.ts:23]: done

File: pages/api/create-grant.ts (hunk 3 above)
Comment (rachelweinberg12) [reply pages/api/create-grant.ts]: Ohh done

**Comment by akrolsmir** [comment]
Looks good to me(rge)!

**Pull request merged**